writer.go

```diff
@@ -585,15 +611,30 @@ func (w *writer) run() {
 
 	for !done {
 		var mustFlush bool
-
+		// lstMsg gets set when the next message would put the maxMessageBytes  over the limit.
+		// If a lstMsg exists we need to add it to the batch so we don't lose it.
+		if lstMsg != nil {
```

Comment:
Aren't we potentially losing the `lastMessage` here?  If the `lastMessage` was the last message in the channel and the channel closed then we wouldn't enter into this loop again (`done` would be true).   I might be wrong, it's a little difficult to tell from the PR view.

Reply:
I think that if the chan was closed you would hit https://github.com/newrelic-forks/kafka-go/pull/4/files#diff-dc03dbc53655fac7a4c8d16b1b4144c1R623 which would  set mustFlush and you would produce that final message in a batch.

Reply:
Right, but you never put the `lstMsg` into the the batch in that case.

Reply:
I gets appened right here:
https://github.com/newrelic-forks/kafka-go/pull/4/files#diff-dc03dbc53655fac7a4c8d16b1b4144c1R617

Reply:
ohh, I see what your saying. 
Yeah it needs to be appened in the `!ok` block

Reply:
That seems like a simple solution.